deployment/terraform/assets/cluster.tf

```diff
@@ -82,6 +86,50 @@ resource "aws_instance" "app_server" {
   }
 }
 
+data "aws_vpc" "selected" {
+  tags = {
+    Name = "Default VPC"
+  }
+}
+
+data "aws_subnets" "loadtest_subnets" {
+  filter {
+    name = "vpc-id"
+    values = [data.aws_vpc.selected.id]
+  }
+
+  tags = {
+    Name = "loadtest*"
+  }
+}
```

Comment:
I still think we can use the VPC configured by the user in `var.cluster_vpc_id`. Resources will use the default VPC if their `vpc_id` is empty. Also, we should further investigate whether we can let users define subnets to follow what @fmartingr did in https://github.com/mattermost/mattermost-load-test-ng/pull/833 As before, I think resources will use the default subnet of the chosen VPC if their `subnet_id` is empty.

Reply:
I couldn't find another way of getting subnets without querying the VPC explicitly. Without getting the subnets, I couldn't create the mount targets as they are required. So I had to get the VPC in use that's why we have this section. Sorry, my knowledge on terraform is limited and probably it came to an end here so I don't understand why we would need to define specific subnets separately like you mentioned apart from what we get from the VPC for load testing.

Reply:
@agarciamontoro thanks for the pointer, unfortunately it doesn't get the subnets without explicitly defining the VPC. Or, at least I couldn't find a way to do so. `data.aws_subnets.selected` is empty during the creation of the efs mount targets hence it doesn't create any target and then app servers are failing to start (because I'm trying to mount them during the provisioning). Even querying for the selected vpc didn't work, I don't know why as there is no meaningful error from terraform but only `exit status 1`.

Reply:
Got it! Thanks for the time spent on this, my assumptions were obviously wrong, sorry! We'll have to take a deeper look into this at some point.